core/tee/tee_svc_storage.c

```diff
@@ -705,9 +705,11 @@ TEE_Result syscall_storage_reset_enum(unsigned long obj_enum)
 	if (res != TEE_SUCCESS)
 		return res;
 
-	e->fops->closedir(e->dir);
-	e->fops = NULL;
-	e->dir = NULL;
+	if (e->fops) {
+		e->fops->closedir(e->dir);
+		e->fops = NULL;
+		e->dir = NULL;
```

Comment:
No because if `fops` is NULL, `dir` should also be NULL otherwise there is an inconsistency. I'll add an assertion.